httpcore/_async/connection_pool.py

```diff
@@ -294,3 +295,15 @@ async def aclose(self) -> None:
         # Close all connections
         for connection in connections:
             await connection.aclose()
+
+    def get_connection_stats(self) -> Dict[Origin, Dict[str, int]]:
+        """Returns statistics dict around state of the connection pool in the form:
+
+        { origin: { connection_state: number_of_connections_on_that_state } }
+        """
+        stats: Dict[Origin, Dict[str, int]] = {}
+        for origin, connections in self._connections.items():
+            stats[origin] = defaultdict(lambda: 0)
```

Comment:
Could use `collections.Counter`?

Reply:
That was my first idea, but couldn't find a clear way of introspecting the state and counting based on that. Didn't think too hard about it though... 😂